docs/proposals/localDisk_volume.md

```diff
@@ -0,0 +1,309 @@
+	VolumeType string `json:"volumetype.omitempty"`
+	// VolumeName is the binding reference to the PersistentVolume backing this
+	// claim. When set to non-empty value Selector is not evaluated
+	VolumeName string `json:"volumeName,omitempty"`
+	// If Recycle is true, the data on the localDisk will be removed when pod is deleted.
+	// Else the data will be kept after pod is deleted.
+	Recycle bool `json:"recycle,omitempty"`
+}
+
+type LocalVolumeClaimStatus struct {
+	// Phase represents the current phase of LocalVolumeClaim
+	Phase LocalVolumeClaimPhase `json:"phase,omitempty"`
+	// AccessModes contains all ways the volume backing the PVC can be mounted
+	AccessModes []LocalVolumeAccessMode `json:"accessModes,omitempty"`
+	// Represents the actual resources of the underlying volume
+	Capacity api.ResourceList `json:"capacity,omitempty"`
+}
+
+type LocalVolumeClaimList struct {
+	unversioned.TypeMeta `json:",inline"`
+	unversioned.ListMeta `json:"metadata,omitempty"`
+	Items                []LocalVolumeClaim `json:"items"`
+}
+
+```
+
+### kubelet enhancement
+
+kubelet will have below changes to support localDisk volume,
+
+* Read config file which has local disks configuration information, format the disks and mount the disks to specified path, e.g. /var/vol-pool/disks/<uuid>, also create LV instances during the kubelet volume plugin initialization.
+
+```
+[disks]
+/dev/vdc
+/dev/vdd
+
+[lvm]
+/dev/ldvg
+
+```
+
+* If the `VolumeSource` is localDisk
+    * When pod is scheduled to the Node, kubelet will check the `LocalDiskType`,
+	    * if the type is `disk`, find an unused LV and get the corret path for plugin.
+	    * if the type is `lvm`, allocate required size logical volume, mount it to specified path, e.g. /var/vol-pool/lvm/<vg-name>/<lv-name>. Get the correct path for plugin and update available size of VG in its LV .
+	    * No matter what type it is, a LVC instance is allocated and bind the LVC to corresponding LV.
+
+    * When pod is deleted, kubelet will check the `LocalDiskType`,
+	    * if the type is `disk`, remove all files under disks.
+	    * if the type is `lvm`, umount the logical volume and delete this logical volume, update available size of VG in its LV.
+	    * No matter what type it is, unbind the LVC and LV and remove its LVC instance.
+	   
+* If the `VolumeSource` is LVC, 
+    * When pod is scheduled to Node, get path for the bound LV for plugin.
+    * When pod is deleted, do nothing.  
+`
+
+### kube-scheduler enhancement
+
+New predicates function will be required for LV resource calculation during pod scheduling, and one cache is created for `localvolumes`.  When the function is called for every Node, it will check items in `localvolumes`
+
+* if request's localDisk volume type is `disk`, it will check available disks in specified Node.
+* if request's localDisk volume type is `lvm`, it will check available VG size in specified Node.
+* if pod's volume is LVC, get the Node from its LV, and bind pod to Node. 

```

Comment:
All this should work even without a new type of volume. 

We should monitor and report disk used by local pvs (https://github.com/kubernetes/kubernetes/issues/3518). Today if you ask for a pvc of 1GB, the dynamic provisioner will give you a cloud disk of exactly 1GB (there are variations of this where it might give you a recycled volume based on policy, but you can't assume you get any more). This is exactly how I would expect an emptydir pv to work for example. 

If the pvc says 1gb, and the pv is emptydir, I get an emptydir directory that's 1gb. Further, if I ask for a certain iops the kubelet/scheduler should coopertae to guarantee it.